src/core.ts

```diff
@@ -261,6 +262,54 @@ export const traverseProps = (
   return false;
 };
 
+export type ThenableData =
+  | ({ status: 'fulfilled'; value: unknown } & Thenable<unknown>)
+  | ({ status: 'rejected'; reason: unknown } & Thenable<unknown>)
+  | ({ status: 'pending' } & Thenable<unknown>);
+
+interface ThenableState {
+  _debugThenableState: ThenableData[] | { thenables: ThenableData[] };
+}
+
+function isThenableState(dependencies: object): dependencies is ThenableState {
+  return '_debugThenableState' in dependencies;
```

Comment:
is this available in the react production build?

Reply:
oh off, I missed that part. Yeah it's dev-only